src/components/calculator.js

```diff
@@ -1,104 +1,289 @@
 /* eslint-disable react/prefer-stateless-function */
 
 import React from 'react';
+import Button from './button';
+
+import { Calculate, Operate } from '../logic';
+
+const operators = ['%', '+', 'x', '-', '+'];
+
+function isNumber(item) {
+  return !!item.match(/[0-9]+/);
+}
 
 class Calculator extends React.Component {
+  constructor(props) {
+    super(props);
+    this.inputRef = React.createRef();
+
+    this.state = {
+      data: {
+        total: null,
+        next: null,
+        operation: null,
+      },
+    };
+
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  componentDidUpdate() {
+    const { data } = this.state;
+    if (this.inputRef.current) {
+      this.inputRef.current.value = `${data.total === null ? '0' : data.total} ${data.operation === null ? '' : data.operation} ${data.next === null ? '' : data.next}`;
+    }
+  }
+
+  handleClick(button) {
+    const stateObj = this.state;
+    const operator = operators.includes(button);
+    let total;
+
+    // if (button === '.') {
+    //   this.setState({
+    //     data: { ...stateObj.data },
+    //   });
+    //   console.log(result);
+    // }
```

Comment:
- [ ] Please kindly remove `commented code` from your code base, as this makes your code base dirty and less readable.